libvirt/domain.go

```diff
@@ -29,6 +30,18 @@ const domWaitLeaseDone = "all-addresses-obtained"
 
 var errDomainInvalidState = errors.New("invalid state for domain")
 
+// TerraformInstanceXML type
+type TerraformInstanceXML struct {
```

Comment:
Regarding this and all functions, I'm thinking that it could make sense to create a domain_tags.go file even if small to don't overload the domaingo imho

Reply:
It could make more sense to maybe call it metadata instead and add to it later maybe? Right now I only added tags to the metadata but in the future other options that need to be saved could also be added to the metadata instance xml.

Reply:
I added a commit with my suggestion.